Assets/Scripts/Buddy/BuddyStats.cs

```diff
@@ -224,7 +223,11 @@ public void AdjustHappiness( float deltaHappiness )
 
 		_happiness = Mathf.Clamp( _happiness, 0f, 1f );
 
-		actor.animator.SetFloat( "happiness", _happiness );
+		// Hidden buddy is disabled
+		if ( gameObject.activeSelf )
+		{
+			actor.animator.SetFloat( "happiness", _happiness );
+		}
```

Comment:
What is this check here needed for? Wouldn't we still want to update the animator's variables for when the buddy becomes visible again?